fraud/subscription.go

```diff
@@ -33,3 +33,28 @@ func (s *subscription) Proof(ctx context.Context) (Proof, error) {
 func (s *subscription) Cancel() {
 	s.subscription.Cancel()
 }
+
+// OnProof accepts the subscription and waiting for the next Fraud Proof.
+// In case if Fraud Proof is received, then handleF will be invoked.
+func OnProof(ctx context.Context, subscription Subscription, handle func(ctx context.Context) error) {
+	var err error
+	defer func() {
+		subscription.Cancel()
+		if err == context.Canceled {
+			return
+		}
+		stopErr := handle(ctx)
+		if stopErr != nil {
+			log.Warn(stopErr)
+		}
+	}()
+	// At this point we receive already verified fraud proof,
+	// so there are no needs to call Validate.
+	_, err = subscription.Proof(ctx)
+	if err != nil {
+		if err == context.Canceled {
+			return
+		}
+		log.Errorw("reading next proof failed", "err", err)
+	}
```

Comment:
As we removed a bunch of stuff out of here, we can now simplify this code a little. 
```suggestion
	// At this point we receive already verified fraud proof,
	// so there are no needs to call Validate.
	_, err := subscription.Proof(ctx)
	if err != nil {
		if err != context.Canceled {
			log.Errorw("reading next proof failed", "err", err)
		}
		return
	}
	
	err := handle(ctx)
	if err != nil {
		log.Warn(stopErr)
	}
```

Also, this func should not `Cancel` the subscription because it does not create it.